libevmasm/Assembly.cpp

```diff
@@ -1156,6 +1272,166 @@ LinkerObject const& Assembly::assemble() const
 	return ret;
 }
 
+LinkerObject const& Assembly::assembleEOF() const
+{
+	LinkerObject& ret = m_assembledObject;
+
+	std::map<size_t, size_t> containerSizes;
+	for (size_t i = 0; i < m_subs.size(); ++i)
+	{
+		auto const& sub = m_subs[i];
+		auto const& linkerObject = sub->assemble();
+		solAssert(linkerObject.immutableReferences.empty());
+		containerSizes[i] = linkerObject.bytecode.size();
+	}
+
+	assertThrow(!m_codeSections.empty(), AssemblyException, "Expected at least one code section.");
+	assertThrow(
+		m_codeSections.front().inputs == 0 && m_codeSections.front().outputs == 0x80, AssemblyException,
+		"Expected the first code section to have zero inputs and be non-returning."
+	);
+
+	std::vector<size_t> codeSectionSizeOffsets;
+	auto setCodeSectionSize = [&](size_t _section, size_t _size) {
+		toBigEndian(_size, bytesRef(ret.bytecode.data() + codeSectionSizeOffsets.at(_section), 2));
```

Comment:
done

Reply:
Because `setBigEndian` checks this.

Reply:
Asserts in `appendBigEndianUint16` are also not needed because `appendBigEndian` also calls `setBigEndian` which check the value to set already. Moreover `toBigEndian` also verifies that value is unsigned.

Reply:
Ah, ok. So it would be best to remove them from `appendBigEndianUint16()`. It confused me because I naturally assumed, they should be in neither or in both so if that's not the case, it's probably wrong :)